app/code/core/Mage/CatalogIndex/Model/Data/Abstract.php

```diff
@@ -118,6 +123,7 @@ public function getParentProductIds($store, $childIds)
      * @param array $settings
      * @param int $type
      * @param int|array $suppliedId
+     * @return mixed
```

Comment:
it should be array as fetchLinkInformation returns array (it has a wrong doc comment, but it always returns $this->_getWriteAdapter()->fetchCol($this->_getLinkSelect()); and fetch call returns an array

Reply:
@tmotyl ... `getParentProductIds()` return either `false` or `fetchLinkInformation()`, that returns `array` indeed.

So it should be `array|false`? Can you please approve and add a note to #784? `mixed` will be checked later!

Reply:
I think you're looking at wrong method.
This method `fetchLinkInformation` has only one return `        return $this->getResource()->fetchLinkInformation($store->getId(), $settings['table'], $idField, $whereField, $suppliedId, $additional);`
Which always returns an array.
Please fix it in this patch.

The other function `getChildProductIds` (which also has "mixed" return type, should have, as you said "array|false"
for this issue I've added a coment in #784 as can be fixed later.